config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_dnsrecordsets.dns.cnrm.cloud.google.com.yaml

```diff
@@ -57,11 +57,17 @@ spec:
           metadata:
             type: object
           spec:
+            not:
```

Comment:
yeah the `not: required:` is redundant given the `oneOf`. I think it gets added by the first resource override ( specifically `PreserveMutuallyExclusiveNonReferenceField`). should we try to figure out if it's ok to remove that?

Reply:
@diviner524 yes, it looks redundant since we already has the oneOf condition.

Though just to confirm: `routingPolicy` field is required when `rrdatas`/`rrdatasRefs` is not set?

@hkundag `PreserveMutuallyExclusiveNonReferenceField` was only designed to handle the ref and nonRef field pairs like `rrdatas` and `rrdatasRefs`. We'll probably need a different function on top of it to handle the general mutual exclusion scenarios for multiple fields.

Reply:
> Though just to confirm: `routingPolicy` field is required when `rrdatas`/`rrdatasRefs` is not set?

yes, in the actual API, exactly one of `routingPolicy` and `rrdatas` must be specified.

the `enforceMutuallyExclusiveRrdatasAndRoutingPolicy` override covers the mutual exclusion for these three fields. Is there a better way with less redundancy? Maybe within `preserveRrdatasFieldAndEnsureRrdatasRefsFieldIsMultiKind`, we can also remove the `not` clause after `PreserveMutuallyExclusiveNonReferenceField`?